gmcs/linglib/lexicon.py

```diff
@@ -1080,22 +1097,16 @@ def validate_lexicon(ch, vr):
 
     # LTX 2022-05-16: Check if there are duplicated features (maybe with different values)
     # with inheritance hierarchy (see discussion in issue #627)
-
-    # Only check if there is no cycle in the hierarchy
-    if not contain_cycle:
-        for lextype in ALL_LEX_TYPES:
-            for lt in ch.get(lextype):
-                # If lt has supertypes
-                lt_supertypes = lt.get('supertypes')
-                if len(lt_supertypes) != 0:
-                    # parent_features_list stores all features from child node (exclusive) to most top parent node
-                    parent_features_list = []
-                    recursive_get_supertypes_features(ch, lt_supertypes, parent_features_list)
-                    # check if the child node has duplicated feature type in parent_features_list
-                    for feat in lt.get('feat'):
-                        for parent_feat in parent_features_list:
-                            if feat.get('name') == parent_feat.get('name'):
-                                mess = 'A value for this feature is already specified on a super type for this ' \
-                                       + lextype + ' class. If the value specified here is conflict with that other' \
-                                              ' specification, the grammar will not compile.'
-                                vr.warn(feat.full_key+'_name', mess)
+    # Abhinav 2023-01-31: Fixed bug blocking multiple inheritance, no longer recursive
+    # so no need to check for cycles (see discussion in issue #671)
+    for lextype in ALL_LEX_TYPES:
+        for lt in ch.get(lextype):
+            feat_name2full_key = {feat.get('name') : feat.full_key for feat in lt.get('feat') if feat}
+            supertype_feat_names = { feat.get('name', '') for feat in get_all_supertypes_features(ch, lt) }
+            dupe_feats = supertype_feat_names.intersection(feat_name2full_key.keys())
```

Comment:
```suggestion
            feat_name2full_key = {feat.get('name'): feat.full_key for feat in lt.get('feat') if feat}
            supertype_feat_names = {feat.get('name', '') for feat in get_all_supertypes_features(ch, lt)}
            dupe_feats = supertype_feat_names.intersection(feat_name2full_key)
```

1. `dict`s iterate through their keys, so `supertype_feat_names.intersection(feat_name2full_key.keys())` is redundant to `supertype_feat_names.intersection(feat_name2full_key)`.
2. Styling: there shouldn't be a space before the colon on line 1104 or before or after the curly braces on 1105.

Reply:
Thanks! Yes, I had realized that about the dictionary keys and meant to change it, but forgot to before committing/PRing this. Good catch!